### PR TITLE
Add example typescript preact

### PR DESCRIPTION
**Pull request by bnn1**
`preact` folder is server-side routing
`preact-full` folder is client-side routing

### CODE REVIEW

**Comment by brillout** [comment]
I'm actually going to rename the examples `react` to `react-server-routing` and `react-full` to `react-client-routing`; we can already starting following that convention for this PR.

The tests are failing.

**Comment by bnn1** [comment]
@brillout  
Tests are failing because test suit does not know that it needs to install dependencies for added examples (I think). If I skip TS checks my examples aren't being tested for the same reason. 

I couldn't figure out how to add my examples to test suit.

**Comment by brillout** [comment]
``Error: Command `npx tsc --noEmit` failed [cwd: /home/runner/work/vite-plugin-ssr/vite-plugin-ssr/examples/preact-full]``

Make sure `npx tsc --noEmit` doesn't throw when ran at `/examples/preact-full`.

**Comment by brillout** [comment]
- Remove TS altogether
 - Rename directories to `examples/preact-server-routing` and `examples/preact-client-routing`.

**Comment by bnn1** [comment]
@brillout done

File: examples/preact-server-routing/renderer/_default.page.client.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+render()
+
+async function render() {
+  // We do Server Routing, but we can also do Client Routing by using `useClientRouter()`
+  // instead of `getPage()`, see https://vite-plugin-ssr.com/useClientRouter
+  const pageContext = await getPage()
+  const { Page, pageProps } = pageContext
+  const body = document.querySelector('body')
+
+  if (body) {
```
Comment (brillout) [inline examples/preact-server-routing/renderer/_default.page.client.jsx:14]: I believe `body` is always defined.

File: examples/preact-server-routing/server/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// @ts-nocheck
```
Comment (brillout) [inline examples/preact-server-routing/server/index.js:1]: We don't need this anymore

**Review by brillout — CHANGES_REQUESTED** [review]

**Comment by brillout** [comment]
Let's see if the boilerplate test run passes.

**Review by brillout — APPROVED** [review]

**Pull request merged**